### ISSUE TITLE
Alerting: Emails being sent if item recovers before delay has expired

### ISSUE THREAD

**Issue by Mattz0r** [librenms/librenms#3570]
Heyup,

Here is an email I've just received:

" RECOVER:
Severity: critical
Duration: 5m 1s
Alert started: 2016-05-27 17:00:34
Modify Threshold "

In my rule, I have the delay set to 15 minutes, this means the device which alerted must have recovered before the 15 minutes is up.

The "ALERT:" email was never received.

@f0o - Are you aware of this at all?

**Comment by f0o** [comment]
Yeah its by design. To avoid flapping. The delay translate to the minimum time a statechange must pass until a notification is sent

**Comment by f0o** [comment]
//addon: the delay is not reflected in the duration or timestamps

**Comment by Mattz0r** [comment]
So that means we have to get random RECOVER emails, irrespective of whether or not we get an ALERT email? - Kinda waste of email really, but hey ho

**Comment by f0o** [comment]
Not so random. They're issued after the alert cleared + whatever the delay is set to.

**Comment by Mattz0r** [comment]
Yes, but I only want to know about an alert if it goes beyond the delayed time... which surely is what the delay is for?

If a port is doing say, 5MB/s for more than 15 minutes, I want to know about it, but if it recovers before the 15 mins, then no email is required, because it fixed by the time the delay is over.

**Comment by f0o** [comment]
Exactly but that's already what happens.

If the alert recovers before the initial delay passes nothing will be sent out

**Comment by Mattz0r** [comment]
But that's not what is happening here.

I'm getting a RECOVER email only, not an ALERT email then a RECOVER email.

**Comment by f0o** [comment]
What the... can you reproduce that on-demand?

I've a bunch of alerts firing off and recovering withing 2mins again which dont alert me

**Comment by Mattz0r** [comment]
I might be able to do it on demand... - Give me a few.

**Comment by f0o** [comment]
If you can please do and share rules etc so I may attempt it as well

We'll get this sorted!

**Comment by laf** [comment]
Also can't replicate this. Please ensure you have the correct timezone for your local env, php and mysql.

If you still have issues then please provide further info.

**Comment by baggar11** [comment]
I seem to have come across this issue too. I've got LibreNMS sitting in an LXC container on Ubuntu 14.04. I've made sure to set all the time zones as well.

With a global alert with the following:
entity: devices_status ="0"
condition: equals
severity: Critical
max alerts: 1
delay: 10m
interval: 1h

I use the default 5 minute snmp interval too. If I manually power off  and on a system as to manually trigger an alert during an snmp scan, but have the recovery within the 10 minute delay period, I still get both device down and recovery emails.
